### PR TITLE
ci: more platforms for CI, artifacts and binary optimization

### PR DESCRIPTION
**Pull request by ThunderEX**
Hi,

I did some minor improvement to CI:

1. add win32 for CirrusCI and add gccWin build for GitHub Actions,
2. enable artifacts for GitHub Actions Make & Meson builds,
3. for GCC, discard unused functions during linking, which reduce binary size by ~40%,

### CODE REVIEW

**Comment by vonericsen** [comment]
Hi @ThunderEX,

These look like good updates overall!

For the binary file size reduction, I am not familiar with these options and how they work and have started reading up on them. I want to make sure I understand how they work and how they may affect portability and how they may work on different CPUs and what compilers support it (specifically going to check clang). If you know anything good to read about these options you can point me too, I will be happy to use that to learn more about them 😄

File: meson.build [hunk 1]
```diff
@@ -2,6 +2,18 @@ project('openSeaChest', 'c', license: 'MPL-2.0', version: '2.0.2')
 
 c = meson.get_compiler('c')
 
+if meson.get_compiler('c').get_id() == 'gcc'
```
Comment (xbjfk) [inline meson.build:5]: ` meson.get_compiler('c')` is already defined as `c`

**Comment by vonericsen** [comment]
Sorry for the delay looking into this.
I have tested it and it seems to work fine.
I also tested these flags with clang and they work there, at least for Linux. From the limited info I found, these are only support on elf binaries (non-Windows).
Can you enable this for clang too in meson?

**Comment by ThunderEX** [comment]
@vonericsen  Thanks for the tip!
I added it for clang and it did reduce the binary size significantly!

**Comment by vonericsen** [comment]
@ThunderEX,
Thanks for making those changes! I've merged this in and look forward to using it going forward

**Pull request merged**